### PR TITLE
Softbans

### PR DESCRIPTION
**Pull request by ghost**
how else am i supposed to observe my testmerges

:cl: More Robust Than You
admin: Adds softbans, which allow one to observe but not play
/:cl:

### CODE REVIEW

**Comment by praisenarsie** [comment]
opening your pr over and over again isn't gonna get it merged

**Comment by Tlaltecuhtli** [comment]
i ded tag pls

**Comment by MoreRobustThanYou** [comment]
yes but i'm gonna fix shit this time k?

**Comment by lzimann** [comment]
already closed before #33407

**Comment by KorPhaeron** [comment]
@Armhulen @MrStonedOne @ausops

I wouldn't mind having this, howabout you guys?

**Comment by praisenarsie** [comment]
giphy

**Comment by Armhulen** [comment]
yeah it's not even just for this one case it's for joan too

it has a place lets do it

**Comment by MrStonedOne** [comment]
This is missing a couple of ways you can enter the round, and should likely have a prompt if an admin attempt to enter them into the round, and a final measure, is a check in living/login that messages_admins with a easy offer body to ghosts link.

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -477,7 +477,7 @@
 			if(!gametypeCheck.age_check(M.client))
 				continue
 		if(jobbanType)
-			if(jobban_isbanned(M, jobbanType) || jobban_isbanned(M, "Syndicate"))
+			if(jobban_isbanned(M, jobbanType) || antag_banned(M))
 				continue
```
Comment (MrStonedOne) [inline code/__HELPERS/game.dm]: if jobbantype is not set, it should still check softban

Comment (MoreRobustThanYou) [reply code/__HELPERS/game.dm]: Done.

File: code/modules/mob/dead/new_player/new_player.dm [hunk 2]
```diff
@@ -36,16 +36,23 @@
...
-		switch(ready)
-			if(PLAYER_NOT_READY)
-				output += "<p>\[ [LINKIFY_READY("Ready", PLAYER_READY_TO_PLAY)] | <b>Not Ready</b> | [LINKIFY_READY("Observe", PLAYER_READY_TO_OBSERVE)] \]</p>"
-			if(PLAYER_READY_TO_PLAY)
-				output += "<p>\[ <b>Ready</b> | [LINKIFY_READY("Not Ready", PLAYER_NOT_READY)] | [LINKIFY_READY("Observe", PLAYER_READY_TO_OBSERVE)] \]</p>"
-			if(PLAYER_READY_TO_OBSERVE)
-				output += "<p>\[ [LINKIFY_READY("Ready", PLAYER_READY_TO_PLAY)] | [LINKIFY_READY("Not Ready", PLAYER_NOT_READY)] | <b> Observe </b> \]</p>"
+		if(is_softbanned(src))
+			switch(ready)
+				if(PLAYER_READY_TO_OBSERVE)
```
Comment (MrStonedOne) [inline code/modules/mob/dead/new_player/new_player.dm]: This code only controls if they are shown the button.

They could still spoof the ready up link or the join game button and the server would let them join.

PoC: 
1. join server
1. go to byond/cache in your documents
1. find the .html file for the ready window, it should be one of the first ones when sorted by modified 
    1.   (512 users, first find the subfolder for your DS instance inside byond cache, sorting by modified should work)
1. grab the ref for `src` from one of the `src=[0x....];ready=blah` links.
1. in the command bar, type `byond://?src=[ref from previous step];ready=1`
1. wait for game to start

Comment (MoreRobustThanYou) [reply code/modules/mob/dead/new_player/new_player.dm]: fixing rn (adding softban checks to Topic())

**Comment by MoreRobustThanYou** [comment]
ok @MrStonedOne i've added checks to dragging ghosts, rudimentary transformations, and VVing ckeys. It will also not let people without +BAN put softbanned people in mobs.

**Comment by KorPhaeron** [comment]
@MoreRobustThanYou I cant reopen this you did the thing again

**Comment by Ausops** [comment]
>@ing me here thinking I respond

this is fine

**Pull request closed without merging**